sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -546,43 +546,6 @@ joint_matrix<sub_group, int, use::accumulator, msize, nsize> sub_c;
 //Remainder handling
 ```
 
-## Future-looking API
-
-### Memory scope
-The current experimental API uses `joint_` semantics to define the memory scope of the matrix. The long term solution is to use the proposed link:../supported/sycl_ext_oneapi_local_memory.asciidoc[`group_local_memory` extension] to allocate the matrix in local memory associated with a SYCL group as shown in the example below.
-
-
-```c++
-multi_ptr<matrix<T>, address_space::local_space> tA_ptr = group_local_memory<matrix<sub_group, int8_t, tM, tN, use::a>>(sg);
-```
-We did not utilize this extension for this matrix API version because sub-group local memory is not yet well defined in {dpcpp}. Moreover, the representation of this notion in LLVM IR and SPIR-V is not clear yet. 
-
-### WI data to joint matrix mapping coordinates information for piece-wise operations
-The indexing provided inside the `wi_data` class accesses only the portion of the matrix held by the current WI. It is not possible to know the location of this portion in the original matrix.  This coordinates mapping  is implementation defined and changes from one backend to the other. For general piece-wise operations like sum of rows of a matrix, the WI data to joint matrix mapping information is needed to reason about the matrix view.
-Within the joint matrix extension, we want to write, as much as possible, one code to run on different backends. If backend X states that a WI owns one exact row of the matrix for instance, writing the following code will work only on that backend for that version of hardware. If a different hardware and implementation is used, the same WI may own only half of the row if, for example, the SG size increased. 
-
-```c++
-auto data = get_wi_data(sg, C);
-for (int i = 0; i < data.length(); ++i) {
-  sum_of_local_rows[row] += data[i];
-}
-```
-
-We want to keep backward compatibility in the joint matrix code when implementations or hardware change. To that end, instead of hard-coding this mapping, we use general backend and target-agnostic functionality, especially in the JIT compilation mode of SYCL. For this reason we would like to be able to query this mapping so that code does not have to change from one version to the other.
-
-So for the mapping problem, since this mapping is implementation-defined, one of the proposals is to add runtime functions like:
-```c++
-auto data = get_wi_data(sg, C);
-for (int i = 0; i < data.length; ++i) {
-  auto row, col = data[i].get_coord();
-  sum_of_local_rows[row] += data[i];
-}
-```
-
-## TODO List
-- Add WI data to joint matrix mapping coordinates information for piece-wise operations. This will be added as part of the query or new methods to the 'get_wi_data' class. 
```

Comment:
This feature is not possible with the "wmma.mma" ptx instructions that we use in `joint_matrix_mad` and this was a conscious decision for which we investigated the pros and cons. 

I don't think it really makes sense to have this get_coord feature in DPC++ for ext_oneapi_cuda backend at all even though it would be technically possible to force it: the only way we could do this would be to support the arch specific "mma" ptx variants of the mad operations which are tuned for a single Nvidia architecture (sm version) and each one (targetting a given sm version) uses different supported sizes. For this reason the mma variants are only currently used in high level libraries like Cutlass, and it would make sense to use them in libraries like SYCL-BLAS in the same way via inline PTX. They are not exposed in the CUDA runtime API, and I do not think it makes sense to expose them in DPC++ either: I think the "wmma.mma" option that we went for is the correct option here to maximize portability across sm versions and allow a more user friendly API.

Reply:
@gmlueck @JackAKirk this feature is important to make get_wi_data usable in a general way. Mapping is important and this is a feedback we got from our ML users and library developers. 
While I put it in the intel document to kick off the discussion, this should be in the unified document once we agree on arguments and all.
On Nvidia, this is implementable by copy to SLM.

Reply:
Yeah, currently you can deal with it by calling `joint_matrix_store` to copy to SLM. If `get_coord` end up being in `oneapi` namespace, then I guess for portability we'd call `joint_matrix_store` in the nvptx impl of `get_coord`. Might be tricky to do this efficiently I guess, but seems hypothetical atm.